src/poll.rs

```diff
@@ -411,6 +411,9 @@ impl Registry {
     /// The readiness interest for an `event::Source` can be changed at any time
     /// by calling [`reregister`].
     ///
+    /// The event source must **not** have been previously registered with this
```

Comment:
I'm not sure this comment should be included in the PR. I welcome new tests, but I think this comment can add some confusion to `Poll` as it currently reads.

Below in `# Notes`, the caution of registering sources and what that means for the lifetime of the source is already conveyed.

Reply:
Could you point me to those `#Notes`?

Reply:
Yep just below a few lines [here](https://github.com/tokio-rs/mio/pull/1219/files#diff-d7937c2d56ffbacce8c1a6acba567e1cR417)!